### PR TITLE
✨ USD0++ Cauldron

### PR DESCRIPTION
**Pull request by 0xCalibur**

### CODE REVIEW

**Review by 0xmDreamy — APPROVED** [review]

File: script/MagicUSD0ppCauldron.s.sol [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    address collateral;
+    address mim;
+    address box;
+    address safe;
+    address masterContract;
+
+    function deploy() public {
+        mim = toolkit.getAddress("mim");
+        box = toolkit.getAddress("degenBox");
+        collateral = 0xdB36f69B88Ec1388dBFAc90132CD396FD4749963; // MagicUSD0++
```
Comment (0xm3rlin) [inline script/MagicUSD0ppCauldron.s.sol]: Incorrect -> implementation not proxy address of UUPS proxy requires changes of all swappers as well

File: script/MagicUSD0ppCauldron.s.sol [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        uint256 interests,
+        uint256 openingFee,
+        uint256 liquidationFee
+    ) private {
+        ProxyOracle oracle = ProxyOracle(deploy(string.concat(name, "_ProxyOracle"), "ProxyOracle.sol:ProxyOracle"));
+        IOracle impl = IOracle(
+            deploy(
+                string.concat(name, "_ERC4626_ChainlinkOracle"),
+                "ERC4626Oracle.sol:ERC4626Oracle",
+                abi.encode(string.concat(name, "/USD"), collateral, chainlinkAggregator)
```
Comment (0xm3rlin) [inline script/MagicUSD0ppCauldron.s.sol:64]: redeploy necessary

File: script/MagicUSD0ppCauldron.s.sol [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        if (oracle.oracleImplementation() != impl) {
+            oracle.changeOracleImplementation(impl);
+        }
+
+        CauldronDeployLib.deployCauldronV4(
+            string.concat("Cauldron_", name),
+            IBentoBoxV1(box),
+            masterContract,
+            IERC20(collateral),
```
Comment (0xm3rlin) [inline script/MagicUSD0ppCauldron.s.sol:76]: redeploy necessary

File: script/MagicUSD0ppCauldron.s.sol [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+        deploy(
+            string.concat(name, "_MIM_TokenSwapper"),
+            "ERC4626Swapper.sol:ERC4626Swapper",
+            abi.encode(box, collateral, mim)
+        );
+        deploy(
+            string.concat(name, "_MIM_LevTokenSwapper"),
+            "ERC4626LevSwapper.sol:ERC4626LevSwapper",
+            abi.encode(box, collateral, mim)
```
Comment (0xm3rlin) [inline script/MagicUSD0ppCauldron.s.sol:93]: redeploy necessary

File: script/MagicUSD0ppCauldron.s.sol [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        );
+        deploy(
+            string.concat(name, "_DegenBoxERC4626Wrapper"),
+            "DegenBoxERC4626Wrapper.sol:DegenBoxERC4626Wrapper",
+            abi.encode(box, collateral)
```
Comment (0xm3rlin) [inline script/MagicUSD0ppCauldron.s.sol:98]: redeploy necessary

**Review by 0xm3rlin — CHANGES_REQUESTED** [review]

**Review by 0xm3rlin — APPROVED** [review]

**Pull request merged**